services/mbin/mbin.tester.js

```diff
@@ -0,0 +1,63 @@
+import Joi from 'joi'
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+t.create('get magazine subscribers')
+  .get('/magazine@instance.tld.json')
+  .intercept(nock =>
+    nock('https://instance.tld/')
+      .get('/api/magazine/name/magazine')
+      .reply(
+        200,
+        JSON.stringify({
+          subscriptionsCount: 42,
+        }),
+      ),
+  )
+  .expectBadge({
+    label: 'subscribe to magazine@instance.tld',
+    message: '42',
+    color: 'brightgreen',
+  })
+
+t.create('bad server or connection')
+  .get('/magazine@instance.tld.json')
+  .networkOff()
+  .expectBadge({
+    label: 'magazine',
+    message: 'inaccessible',
+    color: 'lightgrey',
+  })
+
+t.create('unknown community')
+  .get('/magazine@instance.tld.json')
+  .intercept(nock =>
+    nock('https://instance.tld/')
+      .get('/api/magazine/name/magazine')
+      .reply(
+        404,
+        JSON.stringify({
+          error: 'couldnt_find_magazine',
+        }),
+      ),
+  )
+  .expectBadge({
+    label: 'magazine',
+    message: 'magazine not found',
+    color: 'red',
+  })
+
+t.create('invalid magazine').get('/magazine.invalid.json').expectBadge({
+  label: 'magazine',
+  message: 'invalid magazine',
+  color: 'red',
+})
+
+t.create('test on real mbin magazine for API compliance')
+  .get('/teletext@fedia.io.json')
+  .timeout(10000)
```

Comment:
Is this timeout needed? I'd hope the API to respond much faster than that, many of our badges are displayed in GitHub READMEs, which have a 4s timeout for requests.

Reply:
No not needed, responded in 208ms when I ran the test, I've removed it.